src/main/java/org/jabref/gui/collab/experimental/ExternalChangeDetailsViewFactory.java

```diff
@@ -0,0 +1,13 @@
+package org.jabref.gui.collab.experimental;
+
+import org.jabref.gui.collab.experimental.entrychange.EntryChangeDetailsView;
+import org.jabref.gui.collab.experimental.entrychange.EntryChangeResolver;
+
+public class ExternalChangeDetailsViewFactory {
+
+    public ExternalChangeDetailsView create(ExternalChangeResolver resolver) {
+        return switch (resolver) {
+            case EntryChangeResolver entryChange -> new EntryChangeDetailsView(entryChange.getName(), null, null);
```

Comment:
Pattern matching with switch is still a preview feature and not yet usable

Reply:
Yeah, I run the app and I got an error that says the same😔. I hate using `instanceof`, it always feels like a hack.

Reply:
I had a similar experience recently...But will probably come with jdk20 finally